src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -109,4 +119,37 @@ public void HideIfVisibleAndShowIfHidden() {
 		}
 	}
 
+	private void loadFromFile() {
+		JFileChooser fileChooser = new JFileChooser();
+		fileChooser.setCurrentDirectory(new File(System.getProperty("user.home")));
+		fileChooser.addChoosableFileFilter(new FileNameExtensionFilter("*.json", "json"));
+		fileChooser.addChoosableFileFilter(new FileNameExtensionFilter("*.xml", "xml"));
+		int result = fileChooser.showOpenDialog(this);
+		if (result == JFileChooser.APPROVE_OPTION) {
+			File selectedFile = fileChooser.getSelectedFile();
+			String fileFormat = null;
+			fileFormat = fileChooser.getSelectedFile().getName().substring(fileChooser.getSelectedFile().getName().lastIndexOf(".") + 1);
+			String formattedData = null;
+			try {
+				formattedData = new Scanner(selectedFile).useDelimiter("\\Z").next(); // 1-liner that reads whole file to string
+			} catch (FileNotFoundException e) {
+				e.printStackTrace();
+			}
+			LoadedCommand myLoadedCommand = null;
+			switch (fileFormat) {
+				case "json":
+					myLoadedCommand = new JsonCommandLoader(selectedFile.getPath(), selectedFile.getName()).loadCommandFromExternalSource(formattedData);
+					break;
+				case "xml":
+					myLoadedCommand = new XMLCommandLoader(selectedFile.getPath(), selectedFile.getName()).loadCommandFromExternalSource(formattedData);
+					break;
+				default:
+					break;
+			}
```

Comment:
This method is to complex.
Extract selecting CommandLoader to a separate file and implement it as a factory (or better cache). Here the code should look like this:
CommandLoader loader = CommandLoaderFactory.getInstance(fileFormat);
LoadedCommand myLoadedCommand = loader.loadCommandFromExternalSource(formattedData);

Reply:
Done.